packages/docusaurus-theme-classic/src/theme/BlogLayout/index.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import clsx from 'clsx';
+
+import Layout from '@theme/Layout';
+import BlogSidebar from '@theme/BlogSidebar';
+import TOC from '@theme/TOC';
+
+import type {Props} from '@theme/BlogLayout';
+
+function BlogLayout(props: Props): JSX.Element {
+  const {sidebar, toc, children, ...layoutProps} = props;
+  const hasSidebar = sidebar && sidebar.items.length > 0;
+
+  return (
+    <Layout {...layoutProps}>
+      <div className="container margin-vert--lg">
+        <div className="row docusaurus-jc-center">
+          {hasSidebar && (
+            <aside className="col col--3">
+              <BlogSidebar sidebar={sidebar!} />
+            </aside>
+          )}
+          <main
+            className={clsx('col', {
+              'col--7': hasSidebar,
+              'col--9': !hasSidebar,
```

Comment:
sidebar is 3 cols so why not go from 7 to 9 ? 

Now without sidebar cols = 11, not 12

If we don't want the content to be 10 (too big?) and keep 9, I suggest to use `hasSidebar ? "col--9 col--offset-1" : "col--7"`=> the main content will be better centered

Reply:
Previously, we used col--7 for content area (main element), so in case of no sidebar exist, increase it to col--9, which is the most appropriate maximum for comfortable reading.

We don't need to use col--offset-1 because we already center all columns via justify-content: center

Reply:
Agree that we'd rather limit the max width

> We don't need to use col--offset-1 because we already center all columns via justify-content: center

It's not exactly the same.
Using `justify-content: center` means the 11 cols will be centered, so there's 1/2 col spacing on both sides but as there's no sidebar, the main content is a bit too much on the left (personal opinion).

Using `col--offset-1` is more akin to using `justify-content: flex-end`, and the main content could become a bit more centered (but this can only be applied to larger desktop IMHO, it would look weird on medium width). What do you think about it?

Reply:
I don't really like this pseudo-centering aligning, but have changed back to this behavior as you proposed.